java/vector/src/main/java/org/apache/arrow/vector/util/ValueVectorUtility.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.util;
+
+import org.apache.arrow.util.Preconditions;
+import org.apache.arrow.vector.ValueVector;
+
+/**
+ * Utility methods for {@link ValueVector}.
+ */
+public class ValueVectorUtility {
+
+  /**
+   * Get the toString() representation of vector suitable for debugging.
+   */
+  public static String getToString(ValueVector vector) {
```

Comment:
To make the values in the middle visible, can we make the API more flexible, by providing this API?

public static String getToString(ValueVector vector, int start, int end)

and call it in ValueVector.toString() like this:

ValueVectorUtility.getToString(this, 0, getValueCount());

Reply:
Thanks for suggestion, I think we could add the API for option, if one want to see middle values, just call this API. But I would prefer call getToString(ValueVector vector) for default for the reason mentioned above (valueCount could be very big) and the behavior is consistent with cpp.
What do you think?

Reply:
> Thanks for suggestion, I think we could add the API for option, if one want to see middle values, just call this API. But I would prefer call getToString(ValueVector vector) for default for the reason mentioned above (valueCount could be very big) and the behavior is consistent with cpp.
> What do you think?

Sure. I think for getToString(ValueVector vector, int start, int end) method, the middle part should also be represented as "...", so as to avoid OOM.
The purpose of getToString(ValueVector vector, int start, int end) method is only to allow users to specify the range at will.

Reply:
I see, fixed as you suggested.